odl/tomo/data/mrc2014.py

```diff
@@ -0,0 +1,197 @@
+|         |       |         |        |identify file type             |
++---------+-------+---------+--------+-------------------------------+
+|54       |213-216|Char     |MACHST  |Machine stamp                  |
++---------+-------+---------+--------+-------------------------------+
+|55       |217-220|Float32  |RMS     |RMS deviation of map from mean |
+|         |       |         |        |density                        |
++---------+-------+---------+--------+-------------------------------+
+"""
+# TODO: add nlabl stuff
+
+MRC_HEADER_BYTES = 1024
+MRC_SPEC_KEYS = {
+    'id': 'Long word',
+    'byte_range': 'Byte',
+    'dtype': 'Data type',
+    'name': 'Name',
+    'description': 'Description'}
+
+# Add more if needed
+MRC_DTYPE_TO_NPY_DTYPE = {
+    'Float32': np.dtype('float32'),
+    'Int32': np.dtype('int32'),
+    'Char': np.dtype('S1')}
+
+MRC_MODE_TO_NPY_DTYPE = {
+    0: np.dtype('uint8'),
+    1: np.dtype('int16'),
+    2: np.dtype('float32'),
+    6: np.dtype('uint16')}
+
+ANGSTROM = 1e-10
+
+
+class FileReaderMRC(FileReaderUncompressedBinary):
+
+    """Reader for the MRC file format(s).
+
+    By default, the MRC2014 format is used, see ``MRC_2014_SPEC`` for
+    details. See also [Che+2015]_ or the `explanations on the CCP4 homepage
+    <http://www.ccpem.ac.uk/mrc_format/mrc2014.php>`_ for the
+    text of the specification.
+
+    References
+    ----------
+    [Che+2015] Cheng, A, Henderson, R, Mastronarde, D, Ludtke, S J,
+    Schoenmakers, R H M, Short, J, Marabini, R, Dallakyan, S, Agard, D,
+    and Winn, M. *MRC2014: Extensions to the MRC format header for electron
+    cryo-microscopy and tomography*. Journal of Structural Biology,
+    129 (2015), pp 146--150.
+    """
+
+    def __init__(self, file, header_fields=None):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        file : file-like or str
+            Stream or filename from which to read the data. The stream
+            is allowed to be already opened in 'rb' mode.
+        header_fields : sequence of dicts, optional
+            Definition of the fields in the header (per row), each
+            containing key-value pairs for the following keys:
+
+            - ``'name'`` : Label for the field.
+            - ``'offset_bytes'`` : Start of the field in bytes.
+            - ``'size_bytes'`` : Size of the field in bytes.
+            - ``'dtype'`` : Data type in Numpy- or Numpy-readable format.
+            - ``'dshape'`` (optional) : The array of values is reshaped to
+              this shape.
+            - ``'description'`` : A human-readable description of the field.
+
+            For the default ``None``, the MRC2014 format is used, see
+            ``MRC2014_SPEC``.
+        """
+        if header_fields is None:
+            header_fields = standardized_fields(
+                spec_fields(MRC_2014_SPEC, id_key=MRC_SPEC_KEYS['id']),
+                keys=MRC_SPEC_KEYS, dtype_map=MRC_DTYPE_TO_NPY_DTYPE)
```

Comment:
This is redundant, see comment [below](https://github.com/odlgroup/odl/pull/716/files#r87861953).